### PR TITLE
collapse R code

### LINKED ISSUE #85: Collapsible Show R Code in the Previewer
The Show R Code take a sufficient part of the Previewer View of a Card, especially if the code is complex.
If Show R Code field will be collapsible and by default hidden then it could look more elegant.

image

### PR DESCRIPTION
**Pull request by Polkas**
closes #85 

blocked by https://github.com/insightsengineering/teal.reporter/pull/161

Add collapse option for r code in the previewer.
The `panel_item` function is copy pasted from teal.widgets, but here not exported.

### CODE REVIEW

File: R/utils.R [hunk 1]
```diff
@@ -9,3 +9,103 @@ get_bs_version <- function() {
...
+#' @param collapsed (`logical`, optional)\cr
+#'  whether to initially collapse panel
+#' @param input_id (`character`, optional)\cr
+#'  name of the panel item element. If supplied, this will register a shiny input variable that
+#'  indicates whether the panel item is open or collapsed and is accessed with `input$input_id`.
+#'
+#' @return (`shiny.tag`)
+#'
+#' @keywords internal
+panel_item <- function(title, ..., collapsed = TRUE, input_id = NULL) {
```
Comment (nikolas-burkoff) [inline R/utils.R:28]: hmm - this seems quite a lot of code to take from I assume teal.widgets? Not sure about this

Comment (Polkas) [reply R/utils.R:28]: There are a few solutions like add shinyBS package dependency to both packages or add teal.widgets dependency to teal.reporter. Both will increase the number of dependencies for our packages.

WHen add teal.widgets here:
```
> setdiff(pacs::pac_deps_user("teal.widgets", local = T)[,1], pacs::pac_deps_user("teal.reporter", local = T)[,1])
 [1] "colorspace"   "fansi"        "farver"       "formatters"   "ggplot2"     
 [6] "gtable"       "isoband"      "labeling"     "lattice"      "MASS"        
[11] "Matrix"       "mgcv"         "munsell"      "nlme"         "pillar"      
[16] "pkgconfig"    "purrr"        "R.cache"      "R.methodsS3"  "R.oo"        
[21] "R.utils"      "RColorBrewer" "rematch2"     "rprojroot"    "rtables"     
[26] "scales"       "shinyjs"      "styler"       "tibble"       "utf8"        
[31] "vctrs"        "viridisLite"
```

Comment (Polkas) [reply R/utils.R:28]: So I will recommend to copy paste this function.

Comment (nikolas-burkoff) [reply R/utils.R:28]: yeah I guess 😞

**Review by nikolas-burkoff — APPROVED** [review]
Yup this does what it says on the tin

**Pull request merged**